cpp/neuralnet/nninputs.h

```diff
@@ -25,23 +25,34 @@ namespace NNPos {
   int getPolicySize(int nnXLen, int nnYLen);
 }
 
+namespace NNInputs {
+  constexpr int SYMMETRY_NOTSPECIFIED = -1;
+  constexpr int SYMMETRY_ALL = -2;
+
+  const int NUM_SYMMETRY_BOOLS = 3;
+  const int NUM_SYMMETRY_COMBINATIONS = 8;
+}
+
 struct MiscNNInputParams {
   double drawEquivalentWinsForWhite = 0.5;
   bool conservativePass = false;
   double playoutDoublingAdvantage = 0.0;
   float nnPolicyTemperature = 1.0f;
   bool avoidMYTDaggerHack = false;
+  // If no symmetry is specified, it will use default or random based on config, unless node is already cached.
+  int symmetry = NNInputs::SYMMETRY_NOTSPECIFIED;
+  // This will get set in initNodeNNOutput if needed, otherwise default to 1 to keep existing behavior within getHash()
+  int numSymmetriesToSample = 1;
 
   static const Hash128 ZOBRIST_CONSERVATIVE_PASS;
   static const Hash128 ZOBRIST_PLAYOUT_DOUBLINGS;
   static const Hash128 ZOBRIST_NN_POLICY_TEMP;
   static const Hash128 ZOBRIST_AVOID_MYTDAGGER_HACK;
+
+  static const Hash128 ZOBRIST_NUM_SYMMETRIES_TO_SAMPLE_HASH[NNInputs::NUM_SYMMETRY_COMBINATIONS];
```

Comment:
this has _HASH while the others don't, inconsistent?

Reply:
I found that all other uses of ZOBRIST in the current code use the name _HASH when it is an array instead of a single value. So I followed that example.